tests/linearizability/failpoints.go

```diff
@@ -99,13 +103,92 @@ func (f killFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.Etcd
 		}
 	}
 
-	err := member.Start(ctx)
+	// get endpoints excluding the killed member client URLs
+	endpoints := make([]string, 0, len(clus.EndpointsV3()))
+	for _, ed := range clus.EndpointsV3() {
+		if ed != member.EndpointsV3()[0] {
+			endpoints = append(endpoints, ed)
+		}
+	}
+
+	if f.waitForSnapshot {
+		// wait for potential leader election
+		time.Sleep(2 * time.Second)
+		cc, err := clientv3.New(clientv3.Config{
+			Endpoints:            endpoints,
+			Logger:               zap.NewNop(),
+			DialKeepAliveTime:    1 * time.Millisecond,
+			DialKeepAliveTimeout: 5 * time.Millisecond,
+			Username:             rootUserName,
+			Password:             rootUserPassword,
+		})
+		require.NoError(t, err)
+		require.NoError(t, addTestUserAuth(ctx, cc))
+		// 5000 is the default number of snapshot count catch up entries
+		// DefaultSnapshotCatchUpEntries = 5000
+		// Try manipulating e2e etcd test binary to 100 locally and snapshot-count here to speed up test
+		require.NoError(t, waitForSnapshot(ctx, cc, 5000, lg))
+		require.NoError(t, cc.Close())
+	}
+
+	return member.Start(ctx)
+}
+
+func addTestUserAuth(ctx context.Context, cc *clientv3.Client) (err error) {
+	if _, err := cc.UserAdd(ctx, testUserName, testUserPassword); err != nil {
+		return err
+	}
+	if _, err := cc.RoleAdd(ctx, testRoleName); err != nil {
+		return err
+	}
+	if _, err := cc.UserGrantRole(ctx, testUserName, testRoleName); err != nil {
+		return err
+	}
+	if _, err := cc.RoleGrantPermission(ctx, testRoleName, "key", "key0", clientv3.PermissionType(clientv3.PermReadWrite)); err != nil {
+		return err
+	}
+	return
+}
+
+func waitForSnapshot(ctx context.Context, cc *clientv3.Client, snapshotCount int64, lg *zap.Logger) error {
+	initialRevision, err := getRevision(ctx, cc)
 	if err != nil {
 		return err
 	}
+	lg.Info("got revision",
+		zap.Int64("initial-revision", initialRevision),
+		zap.Int64("snapshot-catchup-count", snapshotCount),
+	)
+	for {
+		time.Sleep(time.Second)
```

Comment:
nit: this doesn't abide 1s triggerTimeout.

Reply:
ack